src/pymovements/events/engbert.py

```diff
@@ -0,0 +1,199 @@
+"""
+This module holds the implementation for the Engbert microsaccades algorithm.
+"""
+
+
+from __future__ import annotations
+
+import numpy as np
+
+from pymovements.events import Event
+
+
+def microsaccades(
+    positions: np.ndarray,
+    velocities: np.ndarray,
+    threshold: tuple[float] | None = None,
+    threshold_factor: float = 6,
+    threshold_method: str = 'engbert2015',
+    min_duration: int = 6,
+    min_threshold: float = 1e-10,
+) -> list[Event]:
+    """
+    Compute (micro-)saccades from raw samples
+    adopted from Engbert et al Microsaccade Toolbox 0.9
+
+    Parameters
+    ----------
+    positions : np.ndarray, shape (N, 2)
+        x und y coordinates of N samples in chronological order
+    velocities : np.ndarray, shape (N, 2)
+        x and y velocities of N samples in chronological order
+    threshold : float
+         if None: data-driven velocity threshold; if tuple of
+        floats: used to compute elliptic threshold
+    threshold_factor : float
+        factor for relative velocity threshold computation
+    threshold_method : str
+        If no threshold given, compute it with this method.
+    min_duration : int
+        minimal saccade duration in samples
+    min_threshold : float
+        minimal threshold value. Raises ValueError if calculated threshold is too low.
+
+    Returns
+    -------
+        list of arrays of shape (7,): (1) saccade onset, (2) saccade
+            offset, (3) peak velocity, (4) horizontal component (dist from first
+            to last sample of the saccade), (5) vertical component,
+            (6) horizontal amplitude (dist from leftmost to rightmost sample),
+            (7) vertical amplitude
+
+    Raises
+    ------
+        ValueError
+            If x.shape and v.shape do not match.
+            If threshold value is below min_threshold value.
+            If passed threshold is not two-dimensional.
+    """
+    positions = np.array(positions)
+    velocities = np.array(velocities)
+
+    if positions.shape != velocities.shape:
+        raise ValueError('x.shape and v.shape do not match')
```

Comment:
x.shape and v.shape are not really defined -- why not use positions.shape and velocities.shape?